### ISSUE TITLE
[dns] "buffer size too small" error

### ISSUE THREAD

**Issue by rade** [zettio/weave#360]
````
--- FAIL: TestDNSServer (1.53s)
	util.go:69: Unexpected error: dns: buffer size too small
		goroutine 74 [running]:
		github.com/zettio/weave/testing.stackTrace(0xa23700, 0x0, 0x0)
			/home/matthias/go/src/github.com/zettio/weave/testing/util.go:78 +0x88
		github.com/zettio/weave/testing.StackTrace(0x0, 0x0)
			/home/matthias/go/src/github.com/zettio/weave/testing/util.go:73 +0x35
		github.com/zettio/weave/testing.Fatalf(0xc20806cd80, 0x863650, 0x14, 0xc2080c3de8, 0x1, 0x1)
			/home/matthias/go/src/github.com/zettio/weave/testing/util.go:69 +0x27
		github.com/zettio/weave/testing.AssertNoErr(0xc20806cd80, 0x2aafcb1cf0f8, 0xc20800a340)
			/home/matthias/go/src/github.com/zettio/weave/testing/util.go:12 +0xea
		github.com/zettio/weave/nameserver.TestDNSServer(0xc20806cd80)
			/home/matthias/go/src/github.com/zettio/weave/nameserver/server_test.go:75 +0xb79
		testing.tRunner(0xc20806cd80, 0xa20338)
			/usr/local/go/src/testing/testing.go:447 +0xbf
		created by testing.RunTests
			/usr/local/go/src/testing/testing.go:555 +0xa8b
DEBUG: 2015/01/23 11:25:53.989234 [zonedb] Looking for address: [10 0 2 1]
FAIL
exit status 1
FAIL	github.com/zettio/weave/nameserver	2.847s
````

This doesn't fail consistently, so clearly it depends on environmental conditions.

**Comment by rade** [comment]
@bboreham mentioned something about our dns server not implementing TCP fallback. Does that mean the above failure may be encountered in ordinary deployments and cause problems for our users?

**Comment by bboreham** [comment]
Weavedns is using `dns.Exchange` which applies a buffer size of 512 bytes.  We could call via a `Client` object to increase this number; as long as we stay under 1400 bytes it should work in most circumstances.

However, it is still technically possible to have a ginormous DNS reply that requires TCP fallback.

**Comment by rade** [comment]
This should have been closed when #361 was merged. Doing so now.
